src/filelock/_unix.py

```diff
@@ -38,9 +38,7 @@ class UnixFileLock(BaseFileLock):
 
         def _acquire(self) -> None:
             ensure_directory_exists(self.lock_file)
-            open_flags = os.O_RDWR | os.O_TRUNC
-            if not Path(self.lock_file).exists():
-                open_flags |= os.O_CREAT
+            open_flags = os.O_RDWR | os.O_TRUNC | os.O_CREAT
```

Comment:
I worry a little that I had to remove this workaround which was added in: #317.

I can't seem to find a way to keep both of these fixes in place.

Reply:
Yeah, I'm hesitant to approve it without it though 🤔